droplets.go

```diff
@@ -229,6 +229,7 @@ type DropletCreateRequest struct {
 	Volumes           []DropletCreateVolume `json:"volumes,omitempty"`
 	Tags              []string              `json:"tags"`
 	VPCUUID           string                `json:"vpc_uuid,omitempty"`
+	WithDropletAgent  *bool                 `json:"with_droplet_agent,omitempty"`
```

Comment:
I don't think this requires the pointer does it? Does the json marshal not omit it appropriately if unset?

Reply:
there's a test case for it, it seems json marshal does omit it if it's nil

Reply:
I think a pointer is correct here. If I understand this feature correctly, there is a difference between sending `"with_droplet_agent": "false"` and omitting it from the request completely. 

The default value for a bool is `false`, so if we have the `omitempty` tag you can never actually send  `"with_droplet_agent": "false"` as json marshal will exclude it. Though if you remove the `omitempty` tag, not setting `WithDropletAgent` explicitly will result in a request that sets `"with_droplet_agent": "false"`

```
func TestDroplets_WithDropletAgentJsonMarshal(t *testing.T) {
	boolF := false
	boolT := true
	tests := []struct {
		in   *DropletCreateRequest
		want string
	}{
		{
			in:   &DropletCreateRequest{Name: "foo", WithDropletAgent: &boolF},
			want: `{"name":"foo","region":"","size":"","image":0,"ssh_keys":null,"backups":false,"ipv6":false,"private_networking":false,"monitoring":false,"tags":null,"with_droplet_agent":false}`,
		},
		{
			in:   &DropletCreateRequest{Name: "foo", WithDropletAgent: &boolT},
			want: `{"name":"foo","region":"","size":"","image":0,"ssh_keys":null,"backups":false,"ipv6":false,"private_networking":false,"monitoring":false,"tags":null,"with_droplet_agent":true}`,
		},
		{
			in:   &DropletCreateRequest{Name: "foo"},
			want: `{"name":"foo","region":"","size":"","image":0,"ssh_keys":null,"backups":false,"ipv6":false,"private_networking":false,"monitoring":false,"tags":null}`,
		},
	}

	for _, tt := range tests {
		got, err := json.Marshal(tt.in)
		if err != nil {
			t.Fatalf("error: %v", err)
		}
		if !reflect.DeepEqual(tt.want, string(got)) {
			t.Errorf("expected: %v, got: %v", tt.want, string(got))
		}
	}
}
```